### PR TITLE
Avatar - to function (performance)

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
Avatar - to function (performance)
I've added a snapshot test for the screen to verify I did not break anything, I figured we can keep it, WDYT?
I saw a significant performance improvement (15-20%) with my tests, please verify you see it too.
WOAUILIB-2913

## Changelog
Avatar - to function (performance)

### CODE REVIEW

File: src/components/avatar/index.tsx [hunk 1]
```diff
@@ -155,176 +154,105 @@ export type AvatarProps = Pick<AccessibilityProps, 'accessibilityLabel'> &
...
+    name,
+    backgroundColor,
+    useAutoColors,
+    autoColorsConfig,
+    containerStyle,
+    onPress,
+    children
+  } = themeProps;
+  const {size: _badgeSize, borderWidth = 0} = badgeProps;
+  const badgeSize = _badgeSize || DEFAULT_BADGE_SIZE;
```
Comment (lidord-wix) [inline src/components/avatar/index.tsx:197]: Consider moving the default value to the destruction (more readable)

File: src/components/avatar/index.tsx [hunk 2]
```diff
@@ -335,92 +263,136 @@ class Avatar extends PureComponent<AvatarProps> {
...
+      return (
+        <Badge testID={`${testID}.onlineBadge`} {...badgeProps} size={badgeSize} containerStyle={_badgePosition}/>
+      );
+    }
+  };
+
+  const ribbon = () => {
+    if (ribbonLabel) {
+      return (
+        <View style={[_ribbonStyle, styles.ribbon, ribbonStyle]}>
```
Comment (lidord-wix) [inline src/components/avatar/index.tsx:330]: you can memoize the style here as well :)

File: src/components/avatar/index.tsx [hunk 3]
```diff
@@ -335,92 +263,136 @@ class Avatar extends PureComponent<AvatarProps> {
...
+        {!_.isUndefined(text) && (
+          <Text numberOfLines={1} style={textStyle} testID={`${testID}.label`}>
+            {text}
+          </Text>
+        )}
+      </View>
+      {image()}
+      {badge()}
+      {_customRibbon()}
+      {ribbon()}
```
Comment (lidord-wix) [inline src/components/avatar/index.tsx:368]: IMO, `renderImage`, `renderBadge`, `renderCustomRibbon`, and `renderRibbon` are better names when we talk about callable functions and not just memoized values.. WDYT?

**Review by lidord-wix — COMMENTED** [review]
Looks good, 
I left few small comments, mostly about naming :)

File: src/components/avatar/index.tsx (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/avatar/index.tsx:197]: It was there, this is actually a (not so readable :/) fix for a test, if the input is `0` the result should be `DEFAULT_BADGE_SIZE`, let me know if you have a better suggestion.

File: src/components/avatar/index.tsx [hunk 4]
```diff
@@ -155,176 +154,105 @@ export type AvatarProps = Pick<AccessibilityProps, 'accessibilityLabel'> &
...
+  const {size: _badgeSize, borderWidth = 0} = badgeProps;
+  const badgeSize = _badgeSize || DEFAULT_BADGE_SIZE;
+
+  useEffect(() => {
+    if (imageSource) {
       LogService.warn('uilib: imageSource prop is deprecated, use source instead.');
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/avatar/index.tsx:204]: I don't expect the `imageSource` to change, this is the equivalent for the usage in the `constructor`.

[the remark this answers is not in the document]
Comment (lidord-wix) [reply src/components/avatar/index.tsx:204]: But the user can pass `imageSource={undefined}` until he loads the data, and then pass `imageSource={someSource}` after the first mount.. we want to catch that and I don't think it's so rare

File: src/components/avatar/index.tsx (hunk 1 above)
Comment (lidord-wix) [reply src/components/avatar/index.tsx:197]: I see, so let's keep it that way

**Review by lidord-wix — APPROVED** [review]

**Pull request merged**